### ISSUE TITLE
Can't render NaN in JSON from a PORO

### ISSUE THREAD

**Issue by Soliah** [rails/rails#15241]
With Rails 4.1.1 and the 4-1-master branch I can't get NaNs rendering in a JSON response from a PORO:

With a fresh Rails app and the following code:

``` ruby
class Post
  def as_json(options = {})
    {
      :value => Float::NAN
    }
  end
end

class PostsController < ApplicationController
  def index
    post = Post.new
    render json: post
  end
end
```

I get:

```
JSON::GeneratorError (818: NaN not allowed in JSON):
  app/controllers/posts_controller.rb:4:in `index'
```

**Comment by voltechs** [comment]
I apologize if this is a question with an obvious answer, but uh, what is the workaround for something like this? I can't even track down where this NaN is coming from... what's the downside to automatically returning nil/null?

**Comment by StayerX** [comment]
You could modify it prior to the serialization: 
```
def custom_field                                                  
  return 0 if object.field.nil?                        
  return 0 if object.field.to_f < 0                    
  split = object.life_expectancy.split("/")                      
  split.first.to_f / split.second.to_f                           
end
```

**Comment by chancancode** [comment]
Sorry for the late response. Is this still not broken on master? I would think that this should "just work" if you [look at the encoder](https://github.com/rails/rails/blob/master/activesupport/lib/active_support/json/encoding.rb#L83).

I would expect the first pass to go down the `else` branch, then on the next pass it would go down the `Hash` branch, and then on the third pass the hash value (NaN) should go down the `Numeric` branch and bottoms out at [`Float#as_json`](https://github.com/rails/rails/blob/master/activesupport/lib/active_support/core_ext/object/json.rb#L105), which does handle `NaN` and `Infinity`.

So perhaps something is not working the way I expect here. It might not end up being a bug, but I would appreciate it if someone could dig in and find out what's causing this.

**Comment by chancancode** [comment]
Ah. @soliah @voltechs @StayerX this is indeed a bug and it was fixed in #26933.
